### PR TITLE
[Puppi] Hadronic tau recovery with protection for fromPV==2 charged particles

### PR DESCRIPTION
**Pull request by nurfikri89**
#### PR description:

(Making this PR on behalf of @abenecke)

It was spotted by the ParticleNet team that the efficiency to identify hadronic tau candidates for PUPPI jets at low pT is significantly lower than for CHS jets (https://indico.cern.ch/event/1300663/#2-particlenet-ak4-ak8-tagger-r). The main reason for this is that PUPPI treats particles with fromPV1&2 different than CHS.  It was found that PUPPI rejects one prong hadronic taus that most of the time has `fromPV==2` and `pT < 20 GeV` (`pT > 20 GeV` are protected by PUPPI already). This PR is supposed to recover the loss of hadronic tau candidates by keeping charged particles with `pT > 4 GeV` and `fromPV==2` while maintaining the jet energy resolution, eff and purity for jets. A comparison of different options was done in this presentation: https://indico.cern.ch/event/1364503/#5-puppi-tau-tune

#### PR validation:

passes the usual runTheMatrix test: `runTheMatrix.py -l limited -i all --ibeos`

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

To be backported to `14_0_X`

### CODE REVIEW

**Comment by mandrenguyen** [comment]
Based on the PR description, I understand why changes are observed for Puppi jets, but were you expecting changes to the vanilla slimmedJets (presumably CHS?), e.g:
[here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-02-22-2300+f9c0ec/61278/validateJR/136.731_RunSinglePh2016B/all_mini_reRECO_step3_inMINIAOD/c_log10patJets_slimmedJetsAK8__reRECO_obj_et.png)

**Comment by nurfikri89** [comment]
@mandrenguyen the plot says its slimmedJetsAK8 so its the AK8 Puppi jets, which we expect some changes.

**Comment by mandrenguyen** [comment]
+reconstruction
Only puppi jets are affected, although I found a bit confusing that slimmedJetsAK8 is done with PUPPI, but slimmedJets (i.e., R=0.4) are not. 
I wonder if we shouldn't always just state the UE subtraction algo for clarity, e.g., slimmedJetsCHS.
At least if we're planning to continue to support two algos.

**Comment by hqucms** [comment]
@nurfikri89 Has it been checked what are impacts of this change on the existing AK4 and AK8 jet taggers?

**Comment by nurfikri89** [comment]
@hu, this is in the process by @abenecke

**Comment by vlimant** [comment]
+1

**Comment by antoniovilela** [comment]
please test
   - need to refresh tests.

**Comment by antoniovilela** [comment]
> @nurfikri89 Has it been checked what are impacts of this change on the existing AK4 and AK8 jet taggers?

@nurfikri89 
You mentioned this was being checked. Could you summarize the conclusions?
Thanks.

**Comment by rappoccio** [comment]
> +reconstruction Only puppi jets are affected, although I found a bit confusing that slimmedJetsAK8 is done with PUPPI, but slimmedJets (i.e., R=0.4) are not. I wonder if we shouldn't always just state the UE subtraction algo for clarity, e.g., slimmedJetsCHS. At least if we're planning to continue to support two algos.

This was a historical accident.

**Comment by rappoccio** [comment]
Apologies, I accidentally clicked the wrong button.

**Comment by rappoccio** [comment]
+1

**Comment by mandrenguyen** [comment]
type tau, jetmet
